### DISCUSSION TITLE
大佬有测试过arm mac上程序的可行性吗

### DISCUSSION THREAD

**Discussion by mobeicanyue** [zdz/ServerStatus-Rust#231] [Q&A]
我写了一个很简单的程序然后github action配置文件是照着你改的，虽然cargo build编译能通过。但是给aarch mac的人测试了好像不能运行。

（aarch apple 没有这个docker 镜像，你可以看看）
所以写的cross 实际上还是cargo编译的
![image](https://github.com/zdz/ServerStatus-Rust/assets/81098819/e83ba7e2-ae95-4672-bd0a-d6dde9de2a35)

**Comment by zdz** [comment]
暂时没机器测试
https://github.com/cross-rs/cross-toolchains#apple-targets
![image](https://github.com/zdz/ServerStatus-Rust/assets/152173/aa60a1c2-a823-4a68-8672-4788287b02c9)

可以自己手动 build 一下

**Reply by mobeicanyue** [reply]
1. action-rs/toolchain 没人维护了 早点迁移吧，替代品有dtolnay/rust-toolchain
2. strip可能会导致不同架构的问题，github action只提供x64的机器，对aarch64的进行strip可能会出问题 虽然没报错。(问的AI)
3. 精简操作在cargo.toml里面加个lto=true应该就行了

**Reply by mobeicanyue** [reply]
https://github.com/actions-rs/toolchain/issues/216

https://github.com/rust-netlink/.github/issues/3

**Reply by zdz** [reply]
你可以先试试，最近一段时间旅游没啥时间

**Comment by mobeicanyue** [comment]


**Reply by mobeicanyue** [reply]
太痛苦了
![image](https://github.com/zdz/ServerStatus-Rust/assets/81098819/3a212986-83d2-46fa-a023-1090857ee80a)

下面这个不知道为啥一直用openssl

**Reply by zdz** [reply]
使用 rustls 就是为了去除 openssl 依赖，`default-tls` 用的是 `native-tls`
